lib/global-styles.php

```diff
@@ -253,64 +267,115 @@ function gutenberg_experimental_global_styles_get_theme() {
 	return $theme_config;
 }
 
+/**
+ * Returns the style features a particular block supports.
+ *
+ * @param array $supports The block supports array.
+ *
+ * @return array Style features supported by the block.
+ */
+function gutenberg_experimental_global_styles_get_supported_styles( $supports ) {
+	$style_features = array(
+		'color'            => array( '__experimentalColor' ),
+		'background-color' => array( '__experimentalColor' ),
+		'background'       => array( '__experimentalColor', 'gradients' ),
+		'line-height'      => array( '__experimentalLineHeight' ),
+		'font-size'        => array( '__experimentalFontSize' ),
+	);
+
+	$supported_features = array();
+	foreach ( $style_features as $style_feature => $path ) {
+		if ( gutenberg_experimental_get( $supports, $path ) ) {
+			$supported_features[] = $style_feature;
+		}
+	}
+
+	return $supported_features;
+}
+
 /**
  * Retrieves the block data (selector/supports).
  *
  * @return array
  */
 function gutenberg_experimental_global_styles_get_block_data() {
-	// TODO: this data should be taken from the block registry.
-	//
-	// At the moment this array replicates the current capabilities
-	// declared by blocks via __experimentalLineHeight,
-	// __experimentalColor, and __experimentalFontSize.
 	$block_data = array(
-		'global'          => array(
+		'global' => array(
 			'selector' => ':root',
-			'supports' => array(), // By being blank, the 'global' section won't output any style yet.
-		),
-		'core/paragraph'  => array(
-			'selector' => 'p',
-			'supports' => array( 'line-height', 'font-size', 'color' ),
-		),
-		'core/heading/h1' => array(
-			'selector' => 'h1',
-			'supports' => array( 'line-height', 'font-size', 'color' ),
-		),
-		'core/heading/h2' => array(
-			'selector' => 'h2',
-			'supports' => array( 'line-height', 'font-size', 'color' ),
-		),
-		'core/heading/h3' => array(
-			'selector' => 'h3',
-			'supports' => array( 'line-height', 'font-size', 'color' ),
-		),
-		'core/heading/h4' => array(
-			'selector' => 'h4',
-			'supports' => array( 'line-height', 'font-size', 'color' ),
-		),
-		'core/heading/h5' => array(
-			'selector' => 'h5',
-			'supports' => array( 'line-height', 'font-size', 'color' ),
-		),
-		'core/heading/h6' => array(
-			'selector' => 'h6',
-			'supports' => array( 'line-height', 'font-size', 'color' ),
-		),
-		'core/columns'    => array(
-			'selector' => '.wp-block-columns',
-			'supports' => array( 'color' ),
-		),
-		'core/group'      => array(
-			'selector' => '.wp-block-group',
-			'supports' => array( 'color' ),
-		),
-		'core/media-text' => array(
-			'selector' => '.wp-block-media-text',
-			'supports' => array( 'color' ),
+			'supports' => array( 'background-color' ),
```

Comment:
This one-liner allows the global context to generate CSS for background-color. Adding it in this PR (which is a refactor to take data from the block.json) because it's so small a change.